Support prefersEphemeralWebBrowserSession and useWebView authorize options

carmenlau opened the pull request:
ref #73

louischan-oursky commented on Sources/Authgear.swift:
```diff
@@ -199,6 +202,9 @@ public class Authgear: NSObject {
     public private(set) var sessionState: SessionState = .unknown
 
+    private var currentWebViewRedirectURI: String?
+    public private(set) var authorizeRedirectionHandler: AuthorizeRedirectionHandler = { _ in }
```
Can it be private?

louischan-oursky commented on Sources/Authgear.swift:
```diff
@@ -7,6 +7,7 @@ import WebKit
 
 public typealias AuthorizeCompletionHandler = (Result<AuthorizeResult, Error>) -> Void
 public typealias ReauthenticateCompletionHandler = (Result<ReauthenticateResult, Error>) -> Void
+public typealias AuthorizeRedirectionHandler = (URL) -> Void
```
Can it be private?

louischan-oursky commented on Sources/Authgear.swift:
```diff
@@ -615,17 +743,22 @@ public class Authgear: NSObject {
+        } else {
+            let handler = self.withMainQueueHandler(handler)
+            self.workerQueue.async {
+                self.authorizeWithSession(options, handler: handler)
```
The suffix WithSession and WithoutSession is not too self-explanatory. I suggest we rename them to `authorizeWithWKWebView` and `authorizeWithASWebAuthenticationSession`. Apply to `reauthenticateXXX` as well.

louischan-oursky reviewed (commented):
Can we emulate prefersEphemeralWebBrowserSession for WKWebView? I tested the following scenario.

1. Login with Facebook Login
2. Logout
3. Login with use another account.
4. Click Facebook Login. This time facebook recognized me as logged in already. I cannot use another account :(

carmenlau commented:
Updates:
- Updated to support `preferPrivateWebBrowserSession` for WKWebView and added the option to the demo app
- changed `preferPrivateWebBrowserSession` and `useWebView` default to `false`

carmenlau commented:
Updated in #75
